Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+{
+}
+
+ITSThresholdCalibrationTask::~ITSThresholdCalibrationTask()
+{
+  delete hSuccessRate;
+
+  for (int iScan = 0; iScan < 3; iScan++) {
+    for (int iLayer = 0; iLayer < 7; iLayer++) {
+      delete hCalibrationLayer[iLayer][iScan];
+      delete hCalibrationRMSLayer[iLayer][iScan];
+      if (iScan == 2) {
+        delete hCalibrationThrNoiseLayer[iLayer];
+        delete hCalibrationThrNoiseRMSLayer[iLayer];
+      }
+    }
+    for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+      delete hCalibrationChipSum[iScan][iBarrel];
+      delete hCalibrationChipAverage[iScan][iBarrel];
+      delete hCalibrationChipCounts[iScan][iBarrel];
+
+      delete hCalibrationRMSChipSum[iScan][iBarrel];
+      delete hCalibrationRMSChipAverage[iScan][iBarrel];
+    }
+  }
+
+  for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+
+    delete hCalibrationThrNoiseChipSum[iBarrel];
+    delete hCalibrationThrNoiseChipAverage[iBarrel];
+
+    delete hCalibrationThrNoiseRMSChipSum[iBarrel];
+    delete hCalibrationThrNoiseRMSChipAverage[iBarrel];
+  }
+}
+
+void ITSThresholdCalibrationTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+
+  ILOG(Info, Support) << "initialize ITSThresholdCalibrationTask" << ENDM;
+
+  mRunNumberPath = mCustomParameters["runNumberPath"];
+
+  createAllHistos();
+  publishHistos();
+}
+
+void ITSThresholdCalibrationTask::startOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "startOfActivity" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+
+  ILOG(Info, Support) << "START DOING QC General" << ENDM;
+  const auto tunString = ctx.inputs().get<gsl::span<char>>("tunestring");
+  const auto runType = ctx.inputs().get<short int>("runtype");
+  const auto scanType = ctx.inputs().get<char>("scantype");
+
+  string inString(tunString.begin(), tunString.end());
+
+  Int_t iScan;
+  Double_t CalibrationValue;
+  if (scanType == 'V')
+    iScan = 0;
+  else if (scanType == 'I')
+    iScan = 1;
+  else if (scanType == 'T')
+    iScan = 2;
+
+  auto splitRes = split(inString, "Stave:");
+
+  for (auto StaveStr : splitRes) {
+    if (StaveStr.size() > 0) {
+      CalibrationResStruct result = CalibrationParser(StaveStr);
+
+      int currentStave = StaveBoundary[result.Layer] + result.Stave + 1;
+      int currentChip = result.ChipID + 1;
+      int iBarrel = getBarrel(result.Layer);
+
+      if (scanType == 'V') {
+        CalibrationValue = result.VCASN;
+      } else if (scanType == 'I') {
+        CalibrationValue = result.ITHR;
+      }
+      if (scanType == 'T') {
+        CalibrationValue = result.THR;
+        hCalibrationThrNoiseChipSum[iBarrel]->SetBinContent(currentChip, currentStave, result.Noise + hCalibrationThrNoiseChipSum[iBarrel]->GetBinContent(currentChip, currentStave));
```

Comment:
You can just fill with result.Noise. When a chip arrives in input, it will not reappear again. In other words its noise value is unique.